### DISCUSSION TITLE
Example of how to use vehicle interactions

### DISCUSSION THREAD

**Discussion by pv2-0** [eclipse/mosaic#281] [General]
Hello, 
I am having trouble activating the vehicle distance sensors. It seems like there has been a MOSAIC update and my previous code doesn't run anymore. I was doing something like this in my vehicle application:

getOs().activateVehicleDistanceSensors(4, DistanceSensors.REAR).

From the documentation it looks like I have to use VehicleDistanceSensorActivation but I don't understand how. Should it be used in the onStartup() section of the vehicle application?

I also do not have too much java experience so I may be missing something simple.

Thanks!

**Comment by schwepmo** [comment]
Hello, the API indeed changed a little bit. But you were on the right track initially.
I believe the way we handle `SensorType`s changed a little, no need to use the `VehicleSensorActivation` manually.
Can you try using the following:
```
getOs().activateVehicleSensors(4.0, SensorType.RADAR_REAR);
```

And regarding your second question:
Typically, you would activate the sensors in the `onStartup()`-method to have your vehicle record sensor data from the get-go. However , depending on your use case you can activate/deactivate sensors at any point in your application.

**Reply by pv2-0** [reply]
Thanks!
No matter what I try, I keep getting the error
`package org.eclipse.mosaic.interactions.vehicle.VehicleSensorActivation does not exist`

I need to import that to use SensorType.RADAR_REAR.

Do you have any advice on this? I pulled the updates from github again today just in case but that didn't help.
Any advice would be appreciated.

**Reply by kschrab** [reply]
So you get this error during compilation? Then most likely your project is using outdated or wrong definitions of the required dependencies. What is your pom.xml look like? E.g. it should contain this section:

```xml
<dependencies>
    <dependency>
        <groupId>org.eclipse.mosaic</groupId>
        <artifactId>mosaic-application</artifactId>
        <version>23.0-SNAPSHOT</version>
    </dependency>
</dependencies>
```

**Reply by pv2-0** [reply]
I had 
` <version>21.0</version>` before. Is it okay to just change it manually?

**Reply by pv2-0** [reply]
I changed that line in the pom.xml file and it seems to be fine now. I have other errors to fix but I think they have nothing to do with this.

**Reply by kschrab** [reply]
> I had ` <version>21.0</version>` before. Is it okay to just change it manually?

**The version in your application project should always match with the MOSAIC version you execute your scenario with.** In general, with release versions you should not have any problems in the feature. SNAPSHOT-versions might be unavailable once a release has been published by us (e.g., 22.1-SNAPSHOT wont be available anymore since 22.1 is already released).

**Reply by pv2-0** [reply]
My applications compile fine now that I changed that line in the pom.xml file. I had pulled all updates from github previously, and I checked all the other pom.xml files in all the folders of the program. All of the others had updated to 23.0-SNAPSHOT. However, when I try to run my scenario now, i get the error 
`-NoClassDefFoundError: org/eclipse/mosaic/interactions/vehicle/VehicleSensorActivation$SensorType`
In the MOSAIC.log, I have `ERROR MosaicStarter - Stopping simulation due to a critical error` because it cannot find the VehicleSensorActivation$SensorType class in the vehicle application. This means that something else isn't properly updated, right? Do you have any idea what other things I should check? Thank you!

**Reply by pv2-0** [reply]
I compiled the 23.0 version of mosaic-interactions and I believe that fixed it. I'm now getting an error saying that a different class `org.eclipse.mosaic.interactions.electricit.ChargingDenialResponse is not found`
Is there an easier way to use a simulation that I made a year and a half ago without digging through and renaming things based on the 23.0 variables? It seems like the API changed a lot.

**Reply by schwepmo** [reply]
If you know which version you were using when you setup your scenario, you can download the bundled version [here](https://github.com/eclipse/mosaic/tags) and use the start script `mosaic.sh`  on linux or `mosaic.bat` on windows instead of the integrated `MosaicStarter`. However, I would encourage you to use the newest version as we consistently improve functionality and fix bugs. A proper IDE should show you which parts of the API no longer exist.

As another remark, if you pull changes I would recommend running `mvn clean install` in the root-directory so you don't have divereged versions in different packages.
